xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-livetable/src/main/java/org/xwiki/livedata/internal/livetable/LiveTableRequestHandler.java

```diff
@@ -235,14 +235,23 @@ private void addFilterRequestParameters(Filter filter, Map<String, String[]> req
         List<String> values = filter.getConstraints().stream().filter(Objects::nonNull).map(Constraint::getValue)
             .filter(Objects::nonNull).map(Object::toString).collect(Collectors.toList());
         if (!values.isEmpty()) {
-            requestParams.put(filter.getProperty(), values.toArray(new String[values.size()]));
+            requestParams.put(filter.getProperty(), values.toArray(new String[0]));
```

Comment:
`new String[0]` is also slightly faster, but since this is probably not a measurable change in this case I'll revert.

Reply:
Why do you say it's slightly faster?

Reply:
I agree this is counterintuitive (see https://stackoverflow.com/a/29444594/657524 and https://shipilev.net/blog/2016/arrays-wisdom-ancients/).Quoting so:
> To summarize: the JVM and JIT compiler contains several optimizations that enable it to cheaply create and initialize a new correctly sized array, and those optimizations can not be used if you create the array yourself.

Reply:
Another possibility would be to create an array from the stream instead of a List (using `toArray` instead of `collect`) sice you don't actually need it to be a List ;)